### PR TITLE
[dce] don't process expressions of extern classes (closes #8176)

### LINKED ISSUE #8176: haxe 4 jsrequire
Hi,
with haxe 4 i have notice that extern class with jsrequire arent removed also if i use the macro exclude for eliminate some namespaces. with haxe 3 works but with haxe4 the require is mainteined in the output js

### PR DESCRIPTION
**Pull request by nadako**

### CODE REVIEW

File: tests/misc/projects/Issue8176/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+/main.js
```
Comment (RealyUniqueName) [inline tests/misc/projects/Issue8176/.gitignore]: All files generated into `bin/` are automatically ignored (e.g. `-js bin/main.js`)

**Comment by RealyUniqueName** [comment]
I don't see how that test is related to extern classes.

**Comment by nadako** [comment]
See https://github.com/HaxeFoundation/haxe/issues/8176#issuecomment-485064706

**Comment by RealyUniqueName** [comment]
Ah, ok.

File: tests/misc/projects/Issue8176/.gitignore (hunk 1 above)
Comment (nadako) [reply tests/misc/projects/Issue8176/.gitignore]: Good to know :)

**Review by Simn — APPROVED** [review]

**Pull request merged**